catroid/src/test/java/org/catrobat/catroid/test/content/bricks/UpdateVariableInFormulaBrickTest.java

```diff
@@ -0,0 +1,174 @@
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content.bricks;
+import org.catrobat.catroid.CatroidApplication;
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.WhenScript;
+import org.catrobat.catroid.content.bricks.ConcurrentFormulaHashMap;
+import org.catrobat.catroid.content.bricks.FormulaBrick;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.FormulaElement;
+import org.catrobat.catroid.formulaeditor.UserList;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static org.catrobat.catroid.test.xmlformat.ClassDiscoverer.getAllSubClassesOf;
+import static org.catrobat.catroid.test.xmlformat.ClassDiscoverer.removeAbstractClasses;
+import static org.catrobat.catroid.test.xmlformat.ClassDiscoverer.removeInnerClasses;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class UpdateVariableInFormulaBrickTest {
+
+	private UserList userList;
+	private UserVariable userVariable;
+	private FormulaBrick formulaBrick;
+	private static final String VARIABLE_NAME = "Test";
+	private static final String VARIABLE_NAME_USERVARIABLE_FORMAT = "\"Test\" ";
+	private static final String VARIABLE_NAME_USERLIST_FORMAT = "*Test* ";
+	private static final String INVALID_NAME = "Abcd";
+	private static final String NEW_VARIABLE_NAME = "NewName";
+	private static final String NEW_VARIABLE_USERVARIABLE_FORMAT = "\"NewName\" ";
+	private static final String NEW_VARIABLE_USERLIST_FORMAT = "*NewName* ";
+
+	@Parameterized.Parameters(name = "{0}")
+	public static Iterable<Object[]> data() {
+		List<Object[]> parameters = new ArrayList<>();
```

Comment:
please exclude script bricks that are also formula bricks from this and have them tested in a seperate test class